server/sonar-server/src/main/java/org/sonar/server/user/db/GroupDao.java

```diff
@@ -20,30 +20,45 @@
 
 package org.sonar.server.user.db;
 
+import java.util.Date;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringUtils;
+import org.apache.ibatis.session.RowBounds;
 import org.sonar.api.utils.System2;
+import org.sonar.core.persistence.DaoComponent;
 import org.sonar.core.persistence.DbSession;
 import org.sonar.core.user.GroupDto;
 import org.sonar.core.user.GroupMapper;
-import org.sonar.server.db.BaseDao;
-
-import java.util.List;
 
 /**
  * @since 3.2
  */
-public class GroupDao extends BaseDao<GroupMapper, GroupDto, String> {
+public class GroupDao implements DaoComponent {
+
+  private static final String SQL_WILDCARD = "%";
+  private System2 system;
 
   public GroupDao(System2 system) {
-    super(GroupMapper.class, system);
+    this.system = system;
   }
 
-  @Override
-  protected GroupDto doGetNullableByKey(DbSession session, String key) {
+  public GroupDto selectByKey(DbSession session, String key) {
     return mapper(session).selectByKey(key);
   }
 
-  @Override
-  protected GroupDto doInsert(DbSession session, GroupDto item) {
+  public int countByQuery(DbSession session, @Nullable String query) {
+    return mapper(session).countByQuery(groupSearchToSql(query));
+  }
+
+  public List<GroupDto> selectByQuery(DbSession session, @Nullable String query, int offset, int limit) {
```

Comment:
offset and limit should be together in an object. We have Paging and SearchOptions. I've used SearchOptions, knowing it is supposed to be used in a ES context.